wpas-hl7-translator/src/main/java/wales/nhs/dhcw/inthub/wpasHl7/MprToAdtA40Mapper.java

```diff
@@ -0,0 +1,172 @@
+            adtMessage.getPATIENT().getPID().getPid3_PatientIdentifierList(0).getCx5_IdentifierTypeCode().setValue("NH");
+            adtMessage.getPATIENT().getPID().getPid3_PatientIdentifierList(1).getCx1_IDNumber().setValue(transaction.getUNITNUMBER());
+            adtMessage.getPATIENT().getPID().getPid3_PatientIdentifierList(1).getCx4_AssigningAuthority().getHd1_NamespaceID().setValue(transaction.getSYSTEMID());
+            adtMessage.getPATIENT().getPID().getPid3_PatientIdentifierList(1).getCx5_IdentifierTypeCode().setValue("PI");
+
+            adtMessage.getPATIENT().getPID().getPid5_PatientName(0).getXpn1_FamilyName().getFn1_Surname().setValue(transaction.getSURNAME());
+            adtMessage.getPATIENT().getPID().getPid5_PatientName(0).getXpn2_GivenName().setValue(transaction.getFORENAME());
+            adtMessage.getPATIENT().getPID().getPid5_PatientName(0).getXpn3_SecondAndFurtherGivenNamesOrInitialsThereof().setValue(" ");
+            adtMessage.getPATIENT().getPID().getPid5_PatientName(0).getXpn5_PrefixEgDR().setValue(transaction.getTITLE());
+
+            adtMessage.getPATIENT().getPID().getPid6_MotherSMaidenName(0).getXpn1_FamilyName().getFn1_Surname().setValue(" ");
+            LocalDate date = LocalDate.parse(transaction.getBIRTHDATE(), DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+            String outputDate = date.format(inputdateFormatter);
+            adtMessage.getPATIENT().getPID().getPid7_DateTimeOfBirth().getTs1_Time().setValue(outputDate);//yyyymmdd
+            adtMessage.getPATIENT().getPID().getPid8_AdministrativeSex().setValue(transaction.getSEX());
+            adtMessage.getPATIENT().getPID().getPid9_PatientAlias(0).getXpn1_FamilyName().getFn1_Surname().setValue(" ");
+            adtMessage.getPATIENT().getPID().getPid9_PatientAlias(0).getXpn2_GivenName().setValue(" ");
+
+            //11
+            adtMessage.getPATIENT().getPID().getPid11_PatientAddress(0).getXad1_StreetAddress().getSad1_StreetOrMailingAddress().setValue(transaction.getADDRESS1());
+            adtMessage.getPATIENT().getPID().getPid11_PatientAddress(0).getXad2_OtherDesignation().setValue(transaction.getADDRESS2());
+            adtMessage.getPATIENT().getPID().getPid11_PatientAddress(0).getXad3_City().setValue(transaction.getADDRESS3());
+            adtMessage.getPATIENT().getPID().getPid11_PatientAddress(0).getXad4_StateOrProvince().setValue(transaction.getADDRESS4());
+            adtMessage.getPATIENT().getPID().getPid11_PatientAddress(0).getXad5_ZipOrPostalCode().setValue(transaction.getPOSTCODE());
+            adtMessage.getPATIENT().getPID().getPid11_PatientAddress(0).getXad7_AddressType().setValue("H");
+            adtMessage.getPATIENT().getPID().getPid11_PatientAddress(1).getXad1_StreetAddress().getSad1_StreetOrMailingAddress().setValue(transaction.getCONTACTADDRESS1());
+            adtMessage.getPATIENT().getPID().getPid11_PatientAddress(1).getXad2_OtherDesignation().setValue(transaction.getCONTACTADDRESS2());
```

Comment:
for readability - if we are setting multiple attributes for the same message elements, we can define a variable for an element, eg.:
```java
var patientAddress = adtMessage.getPATIENT().getPID().getPid11_PatientAddress(1);
patientAddress.getXad2_OtherDesignation().setValue(transaction.getCONTACTADDRESS2());
patientAddress.getXad3_City().setValue(transaction.getCONTACTADDRESS3());
...
```